flink-runtime/src/test/java/org/apache/flink/runtime/leaderelection/ZooKeeperLeaderElectionTest.java

```diff
@@ -396,17 +391,22 @@ void testLeaderChangeWriteLeaderInformationOnlyOnce() throws Exception {
      * Tests that the current leader is notified when his leader connection information in ZooKeeper
      * are overwritten. The leader must re-establish the correct leader connection information in
      * ZooKeeper.
+     *
+     * @deprecated This test verifies the change leader information event handling without the
+     *     {@link DefaultLeaderElectionService}. Essentially, it was testing the {@link
+     *     TestingLeaderElectionEventHandler} implementation.
```

Comment:
should we drop it then?

Reply:
Yes, you're right - I removed the test entirely. It's still valid because it tests the communication between the driver implementation and the ZK backend. But this functionality is also tested in [ZooKeeperMultipleComponentLeaderElectionDriverTest#testLeaderInformationChange](https://github.com/apache/flink/blob/6a410899e2c57059f9944e8dd35742efa135838e/flink-runtime/src/test/java/org/apache/flink/runtime/leaderelection/ZooKeeperMultipleComponentLeaderElectionDriverTest.java#L205)